### PR TITLE
Adding typed info for all classes

### LINKED ISSUE #918: Adding types information on the API surface.
Adding types on the public API surface would allow us to do some runtime type checking later on and would allow user's IDE to have more info for static analysis. 

The functions/signatures to type are the ones listed here https://github.com/keras-team/autokeras/blob/master/autokeras/__init__.py

For the context, see #856 where I add some type information on a ImageClassifier.

This issue can be considered easy to solve (good first issue), it's just long to do because the public API surface is big.

I'll do it but some help is welcome :) make sure to make small pull requests.

### PR DESCRIPTION
**Pull request by dineshkumarsarangapani**
### Which issue(s) does this Pull Request fix?

resolves #918 

### Details of the Pull Request

Added typed info for almost all APIs.

### CODE REVIEW

File: autokeras/tuners/greedy.py [hunk 1]
```diff
@@ -200,14 +204,14 @@ def _generate_hp_values(self, hp_names):
...
-        max_trials=10,
-        initial_hps=None,
-        seed=None,
-        hyperparameters=None,
-        tune_new_entries=True,
-        allow_new_entries=True,
+        hypermodel: kerastuner.HyperModel,
+        objective: str = "val_loss",
+        max_trials: int = 10,
+        initial_hps: Optional[List[Dict[str, Any]]] = None,
```
Comment (dineshkumarsarangapani) [inline autokeras/tuners/greedy.py:210]: @haifeng-jin I was not sure on the type for initial_hps the list of dict has str as Key but was able to take Any (bool, float, int, str) as value. Hope you are find with this.

**Comment by dineshkumarsarangapani** [comment]
@haifeng-jin This PR is ready for review. Please review.

File: autokeras/tuners/greedy.py (hunk 1 above)
Comment (haifeng-jin) [reply autokeras/tuners/greedy.py:210]: It think it looks good for now at least. If we have other use cases, we can change it later. Thanks.

**Review by haifeng-jin — APPROVED** [review]
Thanks for the PR! LGTM

**Pull request merged**